src/components/VideoPlayerContexts/VideoPopoverMenuContext.js

```diff
@@ -0,0 +1,78 @@
+import PropTypes from 'prop-types';
+import React, {useCallback, useContext, useMemo, useState} from 'react';
+import _ from 'underscore';
+import * as Expensicons from '@components/Icon/Expensicons';
+import useLocalize from '@hooks/useLocalize';
+import fileDownload from '@libs/fileDownload';
+import * as Url from '@libs/Url';
+import CONST from '@src/CONST';
+import {usePlaybackContext} from './PlaybackContext';
+
+const VideoPopoverMenuContext = React.createContext(null);
+
+function VideoPopoverMenuContextProvider({children}) {
+    const {currentVideoPlayerRef} = usePlaybackContext();
+    const {translate} = useLocalize();
+    const [currentPlaybackSpeed, setCurrentPlaybackSpeed] = useState(CONST.VIDEO_PLAYER.PLAYBACK_SPEEDS[2]);
+
+    const updatePlaybackSpeed = useCallback(
+        (speed) => {
+            setCurrentPlaybackSpeed(speed);
+            currentVideoPlayerRef.current.setStatusAsync({rate: speed});
+        },
+        [currentVideoPlayerRef],
+    );
+
+    const downloadAttachment = useCallback(() => {
+        currentVideoPlayerRef.current.getStatusAsync().then((status) => {
+            const sourceURI = `/${Url.getPathFromURL(status.uri)}`;
+            fileDownload(sourceURI);
+        });
+    }, [currentVideoPlayerRef]);
+
+    const menuItems = useMemo(
+        () => [
+            {
+                icon: Expensicons.Download,
+                text: translate('common.download'),
+                onSelected: () => {
+                    downloadAttachment();
+                },
+            },
+            {
+                icon: Expensicons.Meter,
+                text: translate('videoPlayer.playbackSpeed'),
+                subMenuItems: [
+                    ..._.map(CONST.VIDEO_PLAYER.PLAYBACK_SPEEDS, (speed) => ({
+                        icon: currentPlaybackSpeed === speed ? Expensicons.Checkmark : null,
+                        text: speed.toString(),
+                        onSelected: () => {
+                            updatePlaybackSpeed(speed);
+                        },
```

Comment:
We should have taken care of the highlighting of the sub item once selected. This caused https://github.com/Expensify/App/issues/42425